src/main/java/collab/DataBase.java

```diff
@@ -2,56 +2,69 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.stream.Collectors;
-import collab.Employee;
 
 public class DataBase {
-  private ArrayList<Employee> employeeData = new ArrayList<Employee>();
+  HashSet<String> registerEmployeeNum = new HashSet<String>();
+  public ArrayList<Employee> employeeData = new ArrayList<Employee>();
   
   public ArrayList<Employee> getEmployeeAllData() {
     return employeeData;
   }
   
   public void add(Employee employee) {
-    employeeData.add(employee);        
+    if(registerEmployeeNum.contains(employee.getEmployeeNumber())) {
+      System.out.println(employee.getEmployeeNumber() + "Register Fail!! Employee Data Duplicated!!");
+      return;
+    }
+    registerEmployeeNum.add(employee.getEmployeeNumber());
+    employeeData.add(employee);
   }
-
+  
+  
   public Employee searchItem(String id) {
-    return searchItems("employeeNum", id).get(0);
+    List<Employee> foundItems = searchItems("employeeNum", id); 
+    if(foundItems == null) return null;
+    return foundItems.get(0);
   }
 
   public List<Employee> searchItems(String field, String value) {
       List<Employee> resultItem = new ArrayList<Employee>();
       resultItem.addAll(employeeData.stream()
               .filter(item -> item.getField(field).equals(value))
               .collect(Collectors.toList()));
+      if(resultItem.size()==0) {
+        System.out.println("(field:"+field +" value : "+value+ ") can't find!!");
+        return null;
```

Comment:
이 부분 null로 리턴하니까, 나중에 검색 개수 없을 때 List 객체가 비어 있는 게 아니라 List 자체가 null이라 NullPointException 나더라구요.
임시로 명령어 클래스에서 null일 때마다 ArrayList 만들어서 넣어 주도록 했는데, null 대신에 생성했던 빈 ArrayList 리턴하는 게 좋지 않을까요?


```suggestion
        return resultItem;
```